renderdoc/driver/vulkan/wrappers/vk_sync_funcs.cpp

```diff
@@ -1462,6 +1462,98 @@ void WrappedVulkan::vkCmdWaitEvents2(VkCommandBuffer commandBuffer, uint32_t eve
   }
 }
 
+template <typename SerialiserType>
+bool WrappedVulkan::Serialise_vkTransitionImageLayoutEXT(
+    SerialiserType &ser, VkDevice device, uint32_t transitionCount,
+    const VkHostImageLayoutTransitionInfo *pTransitions)
+{
+  SERIALISE_ELEMENT(device);
+  SERIALISE_ELEMENT(transitionCount);
+  SERIALISE_ELEMENT_ARRAY(pTransitions, transitionCount);
+  if(transitionCount > 0)
+    ser.Important();
+
+  Serialise_DebugMessages(ser);
+
+  SERIALISE_CHECK_READ_ERRORS();
+
+  rdcarray<VkHostImageLayoutTransitionInfo> imgBarriers;
+
+  if(IsReplayingAndReading())
+  {
+    for(uint32_t i = 0; i < transitionCount; i++)
+    {
+      if(pTransitions[i].image != VK_NULL_HANDLE)
+      {
+        imgBarriers.push_back(pTransitions[i]);
+        imgBarriers.back().image = Unwrap(imgBarriers.back().image);
+
+        SanitiseOldImageLayout(imgBarriers.back().oldLayout);
+        SanitiseReplayImageLayout(imgBarriers.back().newLayout);
+
+        if(IsActiveReplaying(m_State) &&
+           m_ReplayOptions.optimisation != ReplayOptimisationLevel::Fastest)
+        {
+          if(pTransitions[i].oldLayout == VK_IMAGE_LAYOUT_UNDEFINED)
+          {
+            VkImageLayout newLayout = imgBarriers.back().newLayout;
+            SanitiseNewImageLayout(newLayout);
+            // TODO: The image is host-writable, so a discard pattern _can_ be written over it;
+            // need to prepare some memory with GetDiscardPattern similar to
+            // VulkanDebugManager::FillWithDiscardPattern() but in memory, and call
+            // vkCopyMemoryToImage.
```

Comment:
This TODO I could probably address without needing help. It can be useful to have this discard pattern if the application initializes the image with `vkTransitionImageLayoutEXT` but then forgets to `vkCopyMemoryToImageEXT` over its contents, or miss parts of the image.